lib/handleRequest.js

```diff
@@ -89,27 +89,57 @@ function preValidationCallback (err, request, reply) {
     return
   }
 
-  const result = validateSchema(reply[kRouteContext], request)
-  if (result) {
-    if (reply[kRouteContext].attachValidation === false) {
-      reply.send(result)
-      return
-    }
+  const validationErr = validateSchema(reply[kRouteContext], request)
+  const isAsync = validationErr && typeof validationErr.then === 'function'
 
-    reply.request.validationError = result
-  }
+  if (isAsync) {
+    validationErr.then(asyncValidationCompleted, asyncValidationCompleted)
+
```

Comment:
Can you move the content of isAsync into a top level function? Function definitions are hoisted at the top of the file.

Reply:
Done